Tests/FoundationTests/FileManagerExtensionsTests.swift

```diff
@@ -80,6 +80,31 @@ final class FileManagerExtensionsTests: XCTestCase {
         } catch {}
     }
 
+    func testDecode() {
+        let point = CGPoint(x: 1, y: 2) //CGPoint conforms to Encodable
+        do {
+            let temporaryDirectoryURL = FileManager.default.temporaryDirectory
+            let temporaryFilename = ProcessInfo().globallyUniqueString
+            let temporaryFileURL = temporaryDirectoryURL.appendingPathComponent(temporaryFilename)
+
+            XCTAssertNoThrow(try JSONEncoder().encode(point))
+            let data1 = try JSONEncoder().encode(point)
```

Comment:
So I was wondering about all these, and I think that if you're doing `try-catch` then you don't need the `XCTAssertNoThrow`s. It seems wasteful having to call all these functions multiple times.
Alternatively I was thinking about having a helper function like this:
```Swift
func XCTAssertNoThrowAndReturn<T>(_ expression: @autoclosure () throws -> T,
                                  _ message: @autoclosure () -> String = "",
                                  file: StaticString = #file,
                                  line: UInt = #line) -> T? {
    do {
        return try expression()
    } catch {
        XCTFail(message())
        return nil
    }
}
```
Then you could just call
```Swift
let data1 = XCTAssertNoThrowAndReturn(try JSONEncoder().encode(point))
```
What do you think?

Reply:
Helper `XCTAssertNoThrowAndReturn` looks a lot cleaner than doing multiple `XCTAssertNoThrow`.

Will this method be useful in future? If you think it's the only case where it is going to be used, then I probably should just get rid of `XCTAssertNoThrow`.

Reply:
As much as I like it, I think it's better the way that you initially did it using do-catch. If you're trying to call and get a value, the next steps probably don't make sense without it if the method you're calling `throw`s. In that case it's fine to let the `do-catch` fail and force the rest of the tests to be skipped. I think `XCTAssertNoThrowAndReturn` may be useful in some scenarios, but perhaps here you don't need it.

Reply:
Sounds good to me. 
Any other changes I should make?

Reply:
Yes, please look at the other comments.